From f5da8d0c38ea0eb660cb17de7aa62cda6899eb75 Mon Sep 17 00:00:00 2001
From: ldv <ldv>
Date: Thu, 1 Jan 2009 23:20:38 +0000
Subject: [PATCH] 2008-11-13  Kirill A. Shutemov  <kirill@shutemov.name>

	* linux/arm/syscallent.h: Fix build on ARM EABI which does not
	provide syscalls socketcall and ipc.
---
 ChangeLog              |    5 +++++
 linux/arm/syscallent.h |    4 +++-
 2 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/linux/arm/syscallent.h b/linux/arm/syscallent.h
index 5ccef2f..edc3641 100644
--- a/linux/arm/syscallent.h
+++ b/linux/arm/syscallent.h
@@ -431,6 +431,7 @@
 	{ 5,	0,	printargs,		"SYS_398"	}, /* 398 */
 	{ 5,	0,	printargs,		"SYS_399"	}, /* 399 */
 
+#ifndef __ARM_EABI__
 #if SYS_socket_subcall != 400
  #error fix me
 #endif
@@ -481,3 +482,4 @@
 	{ 4,	TI,	sys_shmdt,		"shmdt"		}, /* 440 */
 	{ 4,	TI,	sys_shmget,		"shmget"	}, /* 441 */
 	{ 4,	TI,	sys_shmctl,		"shmctl"	}, /* 442 */
+#endif
-- 
1.6.1.3

From 885d06bc92bae2f6aad85ef2fe0b806a06f60712 Mon Sep 17 00:00:00 2001
From: ldv <ldv>
Date: Thu, 1 Jan 2009 23:23:47 +0000
Subject: [PATCH] 2008-11-13  Kirill A. Shutemov  <kirill@shutemov.name>

	* linux/arm/syscallent.h: Update syscalls.
	Based on patch by Enrico Scholz.
---
 ChangeLog              |    3 +
 linux/arm/syscallent.h |   98 ++++++++++++++++++++++++------------------------
 2 files changed, 52 insertions(+), 49 deletions(-)

diff --git a/linux/arm/syscallent.h b/linux/arm/syscallent.h
index edc3641..6dd783a 100644
--- a/linux/arm/syscallent.h
+++ b/linux/arm/syscallent.h
@@ -328,63 +328,63 @@
 	{ 5,	TN,	sys_getsockopt,		"getsockopt"	}, /* 295 */
 	{ 3,	TN,	sys_sendmsg,		"sendmsg"	}, /* 296 */
 	{ 3,	TN,	sys_recvmsg,		"recvmsg"	}, /* 297 */
-	{ 5,	0,	printargs,		"SYS_298"	}, /* 298 */
-	{ 5,	0,	printargs,		"SYS_299"	}, /* 299 */
-	{ 5,	0,	printargs,		"SYS_300"	}, /* 300 */
-	{ 5,	0,	printargs,		"SYS_301"	}, /* 301 */
-	{ 5,	0,	printargs,		"SYS_302"	}, /* 302 */
-	{ 5,	0,	printargs,		"SYS_303"	}, /* 303 */
-	{ 5,	0,	printargs,		"SYS_304"	}, /* 304 */
-	{ 5,	0,	printargs,		"SYS_305"	}, /* 305 */
-	{ 5,	0,	printargs,		"SYS_306"	}, /* 306 */
-	{ 5,	0,	printargs,		"SYS_307"	}, /* 307 */
-	{ 5,	0,	printargs,		"SYS_308"	}, /* 308 */
-	{ 5,	0,	printargs,		"SYS_309"	}, /* 309 */
-	{ 5,	0,	printargs,		"SYS_310"	}, /* 310 */
-	{ 5,	0,	printargs,		"SYS_311"	}, /* 311 */
-	{ 5,	0,	printargs,		"SYS_312"	}, /* 312 */
-	{ 5,	0,	printargs,		"SYS_313"	}, /* 313 */
-	{ 5,	0,	printargs,		"SYS_314"	}, /* 314 */
-	{ 5,	0,	printargs,		"SYS_315"	}, /* 315 */
-	{ 5,	0,	printargs,		"SYS_316"	}, /* 316 */
-	{ 5,	0,	printargs,		"SYS_317"	}, /* 317 */
-	{ 5,	0,	printargs,		"SYS_318"	}, /* 318 */
-	{ 5,	0,	printargs,		"SYS_319"	}, /* 319 */
-	{ 5,	0,	printargs,		"SYS_320"	}, /* 320 */
-	{ 5,	0,	printargs,		"SYS_321"	}, /* 321 */
-	{ 5,	0,	printargs,		"SYS_322"	}, /* 322 */
-	{ 5,	0,	printargs,		"SYS_323"	}, /* 323 */
-	{ 5,	0,	printargs,		"SYS_324"	}, /* 324 */
-	{ 5,	0,	printargs,		"SYS_325"	}, /* 325 */
-	{ 5,	0,	printargs,		"SYS_326"	}, /* 326 */
-	{ 5,	0,	printargs,		"SYS_327"	}, /* 327 */
-	{ 5,	0,	printargs,		"SYS_328"	}, /* 328 */
-	{ 5,	0,	printargs,		"SYS_329"	}, /* 329 */
-	{ 5,	0,	printargs,		"SYS_330"	}, /* 330 */
-	{ 5,	0,	printargs,		"SYS_331"	}, /* 331 */
-	{ 5,	0,	printargs,		"SYS_332"	}, /* 332 */
-	{ 5,	0,	printargs,		"SYS_333"	}, /* 333 */
-	{ 5,	0,	printargs,		"SYS_334"	}, /* 334 */
+	{ 4,	TI,	sys_semop,		"semop"		}, /* 298 */
+	{ 4,	TI,	sys_semget,		"semget"	}, /* 299 */
+	{ 4,	TI,	sys_semctl,		"semctl"	}, /* 300 */
+	{ 4,	TI,	sys_msgsnd,		"msgsnd"	}, /* 301 */
+	{ 4,	TI,	sys_msgrcv,		"msgrcv"	}, /* 302 */
+	{ 4,	TI,	sys_msgget,		"msgget"	}, /* 303 */
+	{ 4,	TI,	sys_msgctl,		"msgctl"	}, /* 304 */
+	{ 4,	TI,	sys_shmat,		"shmat"		}, /* 305 */
+	{ 4,	TI,	sys_shmdt,		"shmdt"		}, /* 306 */
+	{ 4,	TI,	sys_shmget,		"shmget"	}, /* 307 */
+	{ 4,	TI,	sys_shmctl,		"shmctl"	}, /* 308 */
+	{ 5,	0,	printargs,		"add_key"	}, /* 309 */
+	{ 4,	0,	printargs,		"request_key"	}, /* 310 */
+	{ 5,	0,	printargs,		"keyctl"	}, /* 311 */
+	{ 5,	TI,	sys_semtimedop,		"semtimedop"	}, /* 312 */
+	{ 5,	0,	printargs,		"vserver"	}, /* 313 */
+	{ 3,	0,	printargs,		"ioprio_set"	}, /* 314 */
+	{ 2,	0,	printargs,		"ioprio_get"	}, /* 315 */
+	{ 0,	TD,	printargs,		"inotify_init"	}, /* 316 */
+	{ 3,	TD,	sys_inotify_add_watch,	"inotify_add_watch" }, /* 317 */
+	{ 2,	TD,	sys_inotify_rm_watch,	"inotify_rm_watch" }, /* 318 */
+	{ 6,	0,	sys_mbind,		"mbind"		}, /* 319 */
+	{ 3,	0,	sys_set_mempolicy,	"set_mempolicy"	}, /* 320 */
+	{ 5,	0,	sys_get_mempolicy,	"get_mempolicy"	}, /* 321 */
+	{ 4,	TD|TF,	sys_openat,		"openat"	}, /* 322 */
+	{ 3,	TD|TF,	sys_mkdirat,		"mkdirat"	}, /* 323 */
+	{ 4,	TD|TF,	sys_mknodat,		"mknodat"	}, /* 324 */
+	{ 5,	TD|TF,	sys_fchownat,		"fchownat"	}, /* 325 */
+	{ 3,	TD|TF,	sys_futimesat,		"futimesat"	}, /* 326 */
+	{ 4,	TD|TD,	sys_newfstatat,		"newfstatat"	}, /* 327 */
+	{ 3,	TD|TF,	sys_unlinkat,		"unlinkat"	}, /* 328 */
+	{ 4,	TD|TF,	sys_renameat,		"renameat"	}, /* 329 */
+	{ 5,	TD|TF,	sys_linkat,		"linkat"	}, /* 330 */
+	{ 3,	TD|TF,	sys_symlinkat,		"symlinkat"	}, /* 331 */
+	{ 4,	TD|TF,	sys_readlinkat,		"readlinkat"	}, /* 332 */
+	{ 3,	TD|TF,	sys_fchmodat,		"fchmodat"	}, /* 333 */
+	{ 3,	TD|TF,	sys_faccessat,		"faccessat"	}, /* 334 */
 	{ 5,	0,	printargs,		"SYS_335"	}, /* 335 */
 	{ 5,	0,	printargs,		"SYS_336"	}, /* 336 */
-	{ 5,	0,	printargs,		"SYS_337"	}, /* 337 */
-	{ 5,	0,	printargs,		"SYS_338"	}, /* 338 */
-	{ 5,	0,	printargs,		"SYS_339"	}, /* 339 */
-	{ 5,	0,	printargs,		"SYS_340"	}, /* 340 */
+	{ 1,	TP,	sys_unshare,		"unshare"	}, /* 337 */
+	{ 2,	0,	printargs,		"set_robust_list" }, /* 338 */
+	{ 3,	0,	printargs,		"get_robust_list" }, /* 339 */
+	{ 6,	TD,	printargs,		"splice"	}, /* 340 */
 	{ 5,	0,	printargs,		"SYS_341"	}, /* 341 */
-	{ 5,	0,	printargs,		"SYS_342"	}, /* 342 */
-	{ 5,	0,	printargs,		"SYS_343"	}, /* 343 */
+	{ 4,	TD,	printargs,		"tee"		}, /* 342 */
+	{ 4,	TD,	printargs,		"vmsplice"	}, /* 343 */
 	{ 6,	0,	sys_move_pages,		"move_pages"	}, /* 344 */
 	{ 3,	0,	sys_getcpu,		"getcpu"	}, /* 345 */
 	{ 5,	0,	printargs,		"SYS_346"	}, /* 346 */
-	{ 5,	0,	printargs,		"SYS_347"	}, /* 347 */
-	{ 5,	0,	printargs,		"SYS_348"	}, /* 348 */
+	{ 5,	0,	printargs,		"kexec_load"	}, /* 347 */
+	{ 4,	TD|TF,	sys_utimensat,		"utimensat"	}, /* 348 */
 	{ 3,	TD|TS,	sys_signalfd,		"signalfd"	}, /* 349 */
 	{ 4,	TD,	sys_timerfd,		"timerfd"	}, /* 350 */
 	{ 1,	TD,	sys_eventfd,		"eventfd"	}, /* 351 */
-	{ 5,	0,	printargs,		"SYS_352"	}, /* 352 */
-	{ 5,	0,	printargs,		"SYS_353"	}, /* 353 */
-	{ 5,	0,	printargs,		"SYS_354"	}, /* 354 */
+	{ 6,	TF,	sys_fallocate,		"fallocate"	}, /* 352 */
+	{ 4,	TD,	sys_timerfd_settime,	"timerfd_settime"}, /* 353 */
+	{ 2,	TD,	sys_timerfd_gettime,	"timerfd_gettime"}, /* 354 */
 	{ 5,	0,	printargs,		"SYS_355"	}, /* 355 */
 	{ 5,	0,	printargs,		"SYS_356"	}, /* 356 */
 	{ 5,	0,	printargs,		"SYS_357"	}, /* 357 */
-- 
1.6.1.3
